### PR TITLE
Update dart project template

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: lib/src/init.dart [hunk 1]
```diff
@@ -141,15 +141,6 @@ dev_dependencies:
...
-class HelloWorldApp extends App {
-  Widget build() {
-    return new Scaffold(
-        toolbar: new ToolBar(center: new Text("Flutter Demo")),
-        body: new Material(child: new Center(child: new Text("Hello world!"))),
-        floatingActionButton: new FloatingActionButton(
-            child: new Icon(type: 'content/add', size: 24)));
-  }
-}
+void main() => runApp(new Center(child: new Text('Hello, world!')));
```
Comment (abarth) [inline lib/src/init.dart]: Maybe something that builds a Scaffold still?

Comment (devoncarew) [inline lib/src/init.dart]: +1, I do like the look of the current template, even if the code drifted out of date.

**Comment by devoncarew** [comment]
lgtm!

File: lib/src/init.dart [hunk 2]
```diff
@@ -143,13 +143,15 @@ import 'package:sky/widgets.dart';
 
 void main() => runApp(new HelloWorldApp());
```
Comment (abarth) [inline lib/src/init.dart]: I'd actually recommend having this use the `App` widget to get people going in the right direction:

```
void main() {
  runApp(new App(
    title: "Flutter Demo",
    routes: {
     '/': (RouteArguments args) => new FlutterDemoHome()
    }
  ));
}

class FlutterDemoHome extends StatelessComponent { ... }
```

**Comment by chinmaygarde** [comment]
@abarth: Another look?

File: lib/src/init.dart [hunk 3]
```diff
@@ -141,15 +141,26 @@ dev_dependencies:
...
+      routes: <String, RouteBuilder>{
+        '/': (RouteArguments args) => new FlutterDemoHome()
+      }
+    )
+  );
+}
 
-class HelloWorldApp extends App {
-  Widget build() {
+class FlutterDemoHome extends StatelessComponent {
```
Comment (Hixie) [inline lib/src/init.dart]: Call it something else

File: lib/src/init.dart [hunk 4]
```diff
@@ -141,15 +141,26 @@ dev_dependencies:
...
+  Widget build(BuildContext context)  {
     return new Scaffold(
-        toolbar: new ToolBar(center: new Text("Flutter Demo")),
-        body: new Material(child: new Center(child: new Text("Hello world!"))),
-        floatingActionButton: new FloatingActionButton(
-            child: new Icon(type: 'content/add', size: 24)));
+      toolbar: new ToolBar(center: new Text("Flutter Demo")),
+      body: new Material(child: new Center(child: new Text("Hello world!"))),
+      floatingActionButton: new FloatingActionButton(
+        child: new Icon(type: 'content/add', size: 24))
```
Comment (Hixie) [inline lib/src/init.dart]: something's fishy with your parentheses here.

Comment (Hixie) [inline lib/src/init.dart]: Oh, I see. You should put a newline before the last one on the line that says Icon, and then unindent the next line by two spaces

**Comment by Hixie** [comment]
LGTM

**Pull request merged**